### PR TITLE
[PWN-5068] auth

### PR DESCRIPTION
**Pull request by turisap**
🚀 [ JIRA TICKET ](https://p2pvalidator.atlassian.net/browse/PWN-5068)

### CODE REVIEW

File: packages/web/src/new/scenes/Main/Auth/Subviews/Wizard.tsx [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+      </CommonLayout>
+    ),
+    [WizardSteps.CREATE_SET_PASSWORD]: (
+      <CommonLayout>
+        <Password />
+      </CommonLayout>
+    ),
+    [WizardSteps.FINAL]: (
+      <CommonLayout>
+        <p>FINAL</p>
```
Comment (agroznykh) [inline packages/web/src/new/scenes/Main/Auth/Subviews/Wizard.tsx:58]: Is it ok that here we have not Component but just string?

**Review by agroznykh — DISMISSED** [review]

File: packages/web/src/new/scenes/Main/Auth/Subviews/Wizard.tsx [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+export interface Props {
+  step: WizardSteps;
+}
+
+export const Wizard: FC<Props> = observer((props) => {
+  const location = useLocation();
+  const history = useHistory();
+  const viewModel = useViewModel(AuthViewModel);
+
+  // @FIXME remove and replace with react-router
```
Comment (beautyfree) [inline packages/web/src/new/scenes/Main/Auth/Subviews/Wizard.tsx:29]: Should be fix?

File: packages/web/src/App.tsx [hunk 3]
```diff
@@ -45,7 +45,9 @@ const App: React.FC = observer(() => {
               <LocationManager />
               <Switch>
                 <Route path="/" exact component={Landing} />
-                <Route path="/:type(signup|login)" exact component={Auth} />
+                <Route path="/trial" exact component={AuthTrial} />
+                <Route path="/:type(create|restore)" exact component={AuthTrial} />
+                {/*<Route path="/:type(signup|login)" exact component={Auth} />*/}
```
[the remark this answers is not in the document]
Comment (turisap) [reply packages/web/src/App.tsx]: This was a temporary thing for development purposes. I did it to be able to see both new and old auth version.

File: packages/web/src/new/scenes/Main/Auth/Subviews/Wizard.tsx (hunk 2 above)
Comment (turisap) [reply packages/web/src/new/scenes/Main/Auth/Subviews/Wizard.tsx:29]: possibly with the next auth version

File: packages/web/src/new/scenes/Root/Root.ViewModel.ts [hunk 4]
```diff
@@ -1,12 +1,23 @@
...
 
-  protected override onInitialize() {}
+  protected override onInitialize() {
+    this._pricesService.startObserving();
+    this.walletModel.initialize();
+  }
 
-  protected override afterReactionsRemoved() {}
+  protected override afterReactionsRemoved() {
+    this._pricesService.stopObserving();
```
Comment (beautyfree) [inline packages/web/src/new/scenes/Root/Root.ViewModel.ts:21]: should be `this.walletModel.end()`

File: packages/web/src/new/scenes/Main/Settings/Settings.ViewModel.ts [hunk 5]
```diff
@@ -36,6 +40,8 @@ export class SettingsViewModel extends ViewModel {
     void this._nameService
       .getName(this._solanaService.provider.wallet.publicKey.toBase58())
       .then(action((name) => name && (this.username = name)));
+
+    this.walletModel.initialize();
```
Comment (beautyfree) [inline packages/web/src/new/scenes/Main/Settings/Settings.ViewModel.ts:44]: Looks like `this.walletModel.initialize()` is already initilized in `RootViewModel`, so we don't need it again. Correct me if i wrong

File: packages/web/src/new/scenes/Main/Auth/Subviews/Wizard.tsx [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+import { ConfirmMnemonic } from './Create/ConfirmMnemonic';
+import { Mnemonic } from './Create/MnemonicInput';
+import { DerivableAccounts } from './Restore/DerivableAccounts';
+import { RestoreOptions } from './Restore/RestoreOptions';
+
+export interface Props {
+  step: WizardSteps;
+}
+
+export const Wizard: FC<Props> = observer((props) => {
```
Comment (beautyfree) [inline packages/web/src/new/scenes/Main/Auth/Subviews/Wizard.tsx:24]: we already discussed `props` destructurization

File: packages/web/src/new/scenes/Main/Auth/Subviews/components/Final.tsx [hunk 7]
```diff
@@ -85,26 +79,12 @@ const SwitcherText = styled.span`
...
-interface Props {
-  setIsLoading: (isLoading: boolean) => void;
-  data: DataType;
-}
-
-export const Ready: FC<Props> = ({ setIsLoading, data }) => {
-  const { activate } = useWallet();
-  const { setEncryptedSeedAndMnemonic } = useSeedAndMnemonic();
+export const Final: FC = observer(() => {
+  const viewModel = useViewModel(AuthViewModel);
```
[the remark this answers is not in the document]
Comment (turisap) [reply packages/web/src/new/scenes/Main/Auth/Subviews/components/Final.tsx:83]: not sure what you meant here, but `Final` is a part of Auth views, so using `AuthViewModel` should be suitable here

[the remark this answers is not in the document]
Comment (beautyfree) [reply packages/web/src/new/scenes/Main/Auth/Subviews/components/Final.tsx:83]: We should pass `AuthViewModel` from parent (`Auth` component)`

**Pull request merged**